src/test/java/org/datadog/jenkins/plugins/datadog/tracer/TracerInjectionIT.java

```diff
@@ -100,6 +108,18 @@ public void testTracerInjectionInPipelineExecutedOnAgentNode() throws Exception
         }
     }
 
+    @Test
+    public void testTracerInjectionViaPipelineStep() throws Exception {
+        WorkflowJob pipeline = givenPipelineProjectWithTracerEnabledStep();
+        try {
+            givenPipelineIsAMavenBuild(pipeline, false);
+            WorkflowRun build = whenRunningBuild(pipeline);
+            thenTracerIsInjected(build);
+        } finally {
```

Comment:
Could you also add a test where a pipeline is disabling CI visibility but it's enabled globally?

Reply:
There's no way at the moment to enable it globally. It can be enabled in the pipeline/job configuration UI, or by adding the `datadog` step to the pipeline script - both ways only affect a single pipeline.
If it's configured both in the UI and in the script, then the script will have a higher priority (this is how it happens to work, I did not add any reconciliation logic on purpose as I didn't consider it a valid case).